packages/camera/camera_android_camerax/lib/src/android_camera_camerax.dart

```diff
@@ -319,17 +363,101 @@ class AndroidCameraCameraX extends CameraPlatform {
         .bindToLifecycle(cameraSelector!, <UseCase>[preview!]);
   }
 
-  /// Unbinds [preview] instance to camera lifecycle controlled by the
-  /// [processCameraProvider].
-  Future<void> _unbindPreviewFromLifecycle() async {
-    final bool previewIsBound = await processCameraProvider!.isBound(preview!);
-    if (preview == null || !previewIsBound) {
+  /// Configures the [imageAnalysis] instance for image streaming and binds it
+  /// to camera lifecycle controlled by the [processCameraProvider].
+  Future<void> _configureAndBindImageAnalysisToLifecycle() async {
+    assert(processCameraProvider != null);
+    assert(cameraSelector != null);
+
+    if (imageAnalysis != null &&
+        await processCameraProvider!.isBound(imageAnalysis!)) {
+      // imageAnalysis already configured and bound to lifecycle.
       return;
     }
 
+    // Create Analyzer that can read image data for image streaming.
+    Future<void> analyze(ImageProxy imageProxy) async {
+      final List<PlaneProxy> planes = await imageProxy.getPlanes();
+      final List<CameraImagePlane> cameraImagePlanes = <CameraImagePlane>[];
+      for (final PlaneProxy plane in planes) {
+        cameraImagePlanes.add(CameraImagePlane(
+            bytes: plane.buffer,
+            bytesPerRow: plane.rowStride,
+            bytesPerPixel: plane.pixelStride));
+      }
+
+      final int format = imageProxy.format;
+      final CameraImageFormat cameraImageFormat = CameraImageFormat(
+          _imageFormatGroupFromPlatformData(format),
+          raw: format);
+
+      final CameraImageData cameraImageData = CameraImageData(
+          format: cameraImageFormat,
+          planes: cameraImagePlanes,
+          height: imageProxy.height,
+          width: imageProxy.width);
+      cameraImageDataStreamController?.add(cameraImageData);
+      imageProxy.close();
+    }
+
+    final Analyzer analyzer = createDetachedCallbacks
+        ? Analyzer.detached(analyze: analyze)
+        : Analyzer(analyze: analyze);
+
+    // TODO(camsim99): Support resolution configuration.
+    // Defaults to YUV_420_888 image format.
+    imageAnalysis = createImageAnalysis(null);
+    imageAnalysis!.setAnalyzer(analyzer);
+
+    // TODO(camsim99): Reset live camera state observers here when
+    // https://github.com/flutter/packages/pull/3419 lands.
+    camera = await processCameraProvider!
+        .bindToLifecycle(cameraSelector!, <UseCase>[imageAnalysis!]);
+  }
+
+  /// Unbinds [useCase] from camera lifecycle controlled by the
+  /// [processCameraProvider].
+  Future<void> _unbindUseCaseFromLifecycle(UseCase useCase) async {
     assert(processCameraProvider != null);
```

Comment:
FYI you have another assert here followed by a use with !.

Reply:
Going to remove all asserts in this PR.